From 11f031687eb976cf35f3664f0e76c39245b8627c Mon Sep 17 00:00:00 2001
From: Luca Stefani <luca.stefani.ge1@gmail.com>
Date: Tue, 4 Sep 2018 15:05:26 +0200
Subject: [PATCH 59/59] Remove "Avoid NPE during boot phase"

* Commit 8c4a19c fixed them all

Revert "WifiTile: Avoid NPE during boot phase"

This reverts commit 9f03a4ef72bfdb1be85469e47036172364922992.

Revert "DndTile: Avoid yet another NPE during boot up phase"

This reverts commit 8b802eecf4b7e570803e51466afe2f66e0c4c3db.

Revert "DndTile: Avoid NPE during boot phase"

This reverts commit 79ddcfbadc53d42ee8108f6d53ce24b42a747b1c.

Revert "BluetoothTile: Avoid NPE during boot up phase"

This reverts commit 04cfb47d55e1b2171712b932876b4d3e1d9d1070.

Change-Id: Id830df7c67682c7bba290ec89257f70f62bf4c81
---
 .../src/com/android/systemui/qs/tiles/BluetoothTile.java        | 1 -
 .../SystemUI/src/com/android/systemui/qs/tiles/DndTile.java     | 2 --
 .../SystemUI/src/com/android/systemui/qs/tiles/WifiTile.java    | 1 -
 3 files changed, 4 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/qs/tiles/BluetoothTile.java b/packages/SystemUI/src/com/android/systemui/qs/tiles/BluetoothTile.java
index e0ac4683c70..cd0031173f8 100644
--- a/packages/SystemUI/src/com/android/systemui/qs/tiles/BluetoothTile.java
+++ b/packages/SystemUI/src/com/android/systemui/qs/tiles/BluetoothTile.java
@@ -80,7 +80,6 @@ public class BluetoothTile extends QSTileImpl<BooleanState> {
 
     @Override
     public void handleSetListening(boolean listening) {
-        if (mController == null) return;
         if (listening) {
             mController.addCallback(mCallback);
         } else {
diff --git a/packages/SystemUI/src/com/android/systemui/qs/tiles/DndTile.java b/packages/SystemUI/src/com/android/systemui/qs/tiles/DndTile.java
index 2825115f2d6..67900d93be9 100644
--- a/packages/SystemUI/src/com/android/systemui/qs/tiles/DndTile.java
+++ b/packages/SystemUI/src/com/android/systemui/qs/tiles/DndTile.java
@@ -215,7 +215,6 @@ public class DndTile extends QSTileImpl<BooleanState> {
 
     @Override
     protected void handleUpdateState(BooleanState state, Object arg) {
-        if (mController == null) return;
         final int zen = arg instanceof Integer ? (Integer) arg : mController.getZen();
         final boolean newValue = zen != ZEN_MODE_OFF;
         final boolean valueChanged = state.value != newValue;
@@ -278,7 +277,6 @@ public class DndTile extends QSTileImpl<BooleanState> {
     public void handleSetListening(boolean listening) {
         if (mListening == listening) return;
         mListening = listening;
-        if (mController == null) return;
         if (mListening) {
             mController.addCallback(mZenCallback);
             Prefs.registerListener(mContext, mPrefListener);
diff --git a/packages/SystemUI/src/com/android/systemui/qs/tiles/WifiTile.java b/packages/SystemUI/src/com/android/systemui/qs/tiles/WifiTile.java
index 0c796c220b1..d8f7b71d58a 100644
--- a/packages/SystemUI/src/com/android/systemui/qs/tiles/WifiTile.java
+++ b/packages/SystemUI/src/com/android/systemui/qs/tiles/WifiTile.java
@@ -78,7 +78,6 @@ public class WifiTile extends QSTileImpl<SignalState> {
 
     @Override
     public void handleSetListening(boolean listening) {
-        if (mController == null) return;
         if (listening) {
             mController.addCallback(mSignalCallback);
         } else {
-- 
2.17.1

